File: components/offline_items_collection/core/android/java/src/org/chromium/components/offline_items_collection/bridges/OfflineItemVisualsBridge.java
// Copyright 2017 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.components.offline_items_collection.bridges;

import android.graphics.Bitmap;

import org.jni_zero.CalledByNative;
import org.jni_zero.JNINamespace;

import org.chromium.components.offline_items_collection.OfflineItemVisuals;

/**
 * The Java counterpart to the C++ class OfflineItemVisualsBridge
 * (components/offline_items_collection/core/android/offline_item_visuals_bridge.h).  This class has
 * no public members or methods and is meant as a private factory to build
 * {@link OfflineItemVisuals} instances.
 */
@JNINamespace("offline_items_collection::android")
public final class OfflineItemVisualsBridge {
    private OfflineItemVisualsBridge() {}

    /**
     * This is a helper method to allow C++ to create an {@link OfflineItemVisuals} object.
     * @return The newly created {@link OfflineItemVisuals} based on the input parameters.
     */
    @CalledByNative
    private static OfflineItemVisuals createOfflineItemVisuals(Bitmap icon) {
        OfflineItemVisuals visuals = new OfflineItemVisuals();
        visuals.icon = icon;
        return visuals;
    }
}

File: components/offline_items_collection/core/android/offline_item_visuals_bridge.cc
// Copyright 2017 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "components/offline_items_collection/core/android/offline_item_visuals_bridge.h"

#include "components/offline_items_collection/core/offline_item.h"
#include "ui/gfx/android/java_bitmap.h"
#include "ui/gfx/image/image.h"

// Must come after all headers that specialize FromJniType() / ToJniType().
#include "components/offline_items_collection/core/jni_headers/OfflineItemVisualsBridge_jni.h"

using base::android::ScopedJavaLocalRef;

namespace offline_items_collection {
namespace android {

// static
ScopedJavaLocalRef<jobject> OfflineItemVisualsBridge::CreateOfflineItemVisuals(
    JNIEnv* env,
    std::unique_ptr<OfflineItemVisuals> const visuals) {
  if (!visuals)
    return nullptr;

  base::android::ScopedJavaLocalRef<jobject> j_icon;

  if (!visuals->icon.IsEmpty())
    j_icon = gfx::ConvertToJavaBitmap(*visuals->icon.ToSkBitmap());

  return Java_OfflineItemVisualsBridge_createOfflineItemVisuals(env, j_icon);
}

}  // namespace android
}  // namespace offline_items_collection
